server/src/clientglue.js

```diff
@@ -23,6 +23,24 @@ exports.setModel = function (data) {
   model.shot.deleted = data.deleted;
 
   if (firstSet) {
+    if (model.shot.clipNames().length !== 0) {
+      let container = document.querySelector(".clip-container");
+      let img = container.querySelector("img");
+
+      function onResize() {
+        let windowHeight = window.innerHeight;
+        let paddingTop = Math.floor((windowHeight - img.height - 35) / 2);
+        if (paddingTop < 16) {
+          paddingTop = 16;
+        }
+        container.style.paddingTop = paddingTop + "px";
+      }
+      window.addEventListener("resize", onResize, true);
+
+      onResize();
+
+    }
+
```

Comment:
BTW, I have a (rough) lightbox component here: https://github.com/ianb/hashroom/blob/2dbd467cf4d17b371b19f2b85e2ddd87ff1fc7bf/html/js/script.js#L168

It also doesn't deal with resizing, but if I did I'd probably put it in the component onComponentDidMount and hook into window.resize there.  This might require putting the window size in state.